### DISCUSSION TITLE
Is there a tutorial how to update a pip install

### DISCUSSION THREAD

**Discussion by DocMors** [open-webui/open-webui#7261] [General]
Maybe I just didn't find it but can anyone tell me how to update open-webui that was installed via PIP? I reinstalled but it's still many versions behind.
Alternatively an explanation on how to install via GIT would be nice too.

**Comment by carlbeckel** [comment]
I installed mine via pip inside a venv. Here's how I upgrade:
```
> cd <project directory>
> source .venv/bin/activate
> pip install --upgrade pip
> pip install --upgrade open-webui
```

Then leave the venv:
`> deactivate`

Then restart the open-webui daemon/service, which on my Fedora system is:
`> sudo systemctl restart open-webui`

If you can't figure out how to restart the service just reboot the machine.

I just tested these instructions and upgraded to the latest version without issue.

**Reply by DocMors** [reply]
Thanks that helped. I had to twist Debian's arm a little since this is running in a container, so thin it doesn't even use envs. :)

**Comment by mikaelaldy** [comment]
thank you so much, it works on my windows 10
